pkg/sql/doctor/doctor.go

```diff
@@ -46,32 +56,121 @@ func NewDescGetter(rows []DescriptorTableRow) (catalog.MapDescGetter, error) {
 	return pg, nil
 }
 
+func newNamespaceMap(rows []NamespaceTableRow) namespaceReverseMap {
+	res := make(namespaceReverseMap)
+	for _, r := range rows {
+		l, ok := res[r.ID]
+		if !ok {
+			res[r.ID] = []descpb.NameInfo{r.NameInfo}
+		} else {
+			res[r.ID] = append(l, r.NameInfo)
+		}
+	}
+	return res
+}
+
 // Examine runs a suite of consistency checks over the descriptor table.
-func Examine(descTable []DescriptorTableRow, verbose bool, stdout io.Writer) (ok bool, err error) {
-	fmt.Fprintf(stdout, "Examining %d descriptors...\n", len(descTable))
-	descGetter, err := NewDescGetter(descTable)
+func Examine(
+	descTable []DescriptorTableRow,
+	namespaceTable []NamespaceTableRow,
+	verbose bool,
+	stdout io.Writer,
+) (ok bool, err error) {
+	fmt.Fprintf(
+		stdout, "Examining %d descriptors and %d namespace entries...\n",
+		len(descTable), len(namespaceTable))
+	descGetter, err := newDescGetter(descTable)
 	if err != nil {
 		return false, err
 	}
+	nMap := newNamespaceMap(namespaceTable)
+
 	var problemsFound bool
 	for _, row := range descTable {
+		desc, ok := descGetter[descpb.ID(row.ID)]
+		if !ok {
+			// This should never happen as ids are parsed and inserted from descTable.
+			log.Fatalf(context.Background(), "Descriptor id %d not found", row.ID)
+		}
+
+		if int64(desc.GetID()) != row.ID {
+			fmt.Fprint(stdout, reportMsg(desc, "different id in the descriptor: %d", row.ID))
+			problemsFound = true
+			continue
+		}
+
 		table, isTable := descGetter[descpb.ID(row.ID)].(catalog.TableDescriptor)
-		if !isTable {
-			// So far we only examine table descriptors. We may add checks for other
-			// descriptors in later versions.
+		if isTable {
```

Comment:
[nit]: I'd prefer doing a switch on the type here. It would be worthwhile to add validation for types and schemas as well here, since we are just adding them this release

Reply:
Good point - could you describe what needs to be validated for type and schema descriptors?

Reply:
Types have the same validation as tables (parentID and parent schema ID should be valid). Schema descriptors should have parentSchemaID be 0, and a valid parent ID